src/crypto/transactionCache.js

```diff
@@ -132,24 +171,51 @@ export class TransactionCache {
         'getBlockMetadata: undefined vs defined',
       )
       return {
-        lastUpdated: moment(0).toISOString(),
         bestBlockNum: 0,
+        bestBlockHash: null,
+        bestTxHash: null,
       }
     } else {
       // Old addresses
-      assert.assert(
-        metadata.every((x) => x.lastUpdated === first.lastUpdated),
-        'getBlockMetadata: lastUpdated metadata same',
-      )
       assert.assert(
         metadata.every((x) => x.bestBlockNum === first.bestBlockNum),
         'getBlockMetadata: bestBlockNum metadata same',
       )
+      assert.assert(
+        metadata.every((x) => x.bestBlockHash === first.bestBlockHash),
+        'getBlockMetadata: bestBlockHash metadata same',
+      )
 
       return first
     }
   }
 
+  _buildTxHistoryRequest(
+    addresses: Array<string>,
+    metadata: SyncMetadata,
+    currentBestBlockHash: ?string,
+  ): TxHistoryRequest {
+    assert.assert(
+      currentBestBlockHash != null,
+      'buildTxHistoryRequest: bestBlock not null',
+    )
+    /* :: if (currentBestBlockHash == null) throw 'assert' */
```

Comment:
remove  comment?

Reply:
This one is flow comment. We use it in situations in which, even though there is an assert, flow fails to check the type of the test variable.